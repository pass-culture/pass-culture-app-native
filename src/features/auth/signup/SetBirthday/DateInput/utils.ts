export interface DateValidation {
  isComplete: boolean
  isValid: boolean
  isDateAboveMin: boolean
  isDateBelowMax: boolean
}

export interface DateInputRef {
  clearFocuses: () => void
}
