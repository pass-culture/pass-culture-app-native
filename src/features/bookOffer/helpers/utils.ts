import { OfferStockResponse } from 'api/gen'

export enum OfferStatus {
  BOOKABLE = 'BOOKABLE',
  NOT_BOOKABLE = 'NOT_BOOKABLE',
  NOT_OFFERED = 'NOT_OFFERED',
}

export const formatToKeyDate = (dateString: Date | string) => {
  const date = new Date(dateString)
  const day = ('0' + date.getDate()).slice(-2)
  const month = ('0' + (date.getMonth() + 1)).slice(-2)
  const year = date.getFullYear()
  return `${year}-${month}-${day}`
}

export const formatHour = (dateString: string | null | undefined): string => {
  if (!dateString) return ''
  const date = new Date(dateString)
  const hours = date.getHours()
  const minutes = ('0' + date.getMinutes()).slice(-2)
  return `${hours}:${minutes}`
}

export const getStatusFromStocks = (stocks: OfferStockResponse[], credit: number): OfferStatus => {
  const bookableStocks = stocks.filter(({ isBookable }) => isBookable)
  if (bookableStocks.length === 0) return OfferStatus.NOT_BOOKABLE
  return bookableStocks.some(({ price }) => price <= credit)
    ? OfferStatus.BOOKABLE
    : OfferStatus.NOT_BOOKABLE
}
