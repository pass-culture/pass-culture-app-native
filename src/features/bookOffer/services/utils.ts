import { OfferStockResponse } from 'api/gen'
import { formatToSlashedFrenchDate } from 'libs/dates'
import { CENTS_IN_EURO } from 'libs/parsers/pricesConversion'

const EURO_SYMBOL = '€'

export const getStocksByDate = (
  stocks: OfferStockResponse[]
): { [date: string]: OfferStockResponse[] } => {
  const stockDates: { [date: string]: OfferStockResponse[] } = {}

  stocks.forEach((stock) => {
    if (stock.beginningDatetime !== null && stock.beginningDatetime !== undefined) {
      const formattedDate = formatToSlashedFrenchDate(stock.beginningDatetime.toString())
      stockDates[formattedDate] = stockDates[formattedDate]
        ? stockDates[formattedDate].concat(stock)
        : [stock]
    }
  })

  return stockDates
}

export enum OfferStatus {
  BOOKABLE = 'BOOKABLE',
  NOT_BOOKABLE = 'NOT_BOOKABLE',
  NOT_OFFERED = 'NOT_OFFERED',
}

export const formatToFrenchDecimalWithoutSpace = (cents: number) => {
  const euros = cents / CENTS_IN_EURO
  // we show 2 decimals if price is not round. Ex: 21,50€
  const fixed = euros === Math.floor(euros) ? euros : euros.toFixed(2)
  return `${fixed.toString().replace('.', ',')}${EURO_SYMBOL}`
}

export const getDateStatusAndPrice = (
  date: Date,
  stocksDates: { [date: string]: OfferStockResponse[] }
): { status: OfferStatus; price: string | null } => {
  const stocksByDate = stocksDates[formatToSlashedFrenchDate(date.toString())]
  if (!stocksByDate) return { status: OfferStatus.NOT_OFFERED, price: null }

  const prices = stocksByDate.map((stock) => stock.price)
  const price = formatToFrenchDecimalWithoutSpace(Math.min(...prices))
  const offerIsBookable = stocksByDate.some((stock) => stock.isBookable)

  if (offerIsBookable) return { status: OfferStatus.BOOKABLE, price }
  return { status: OfferStatus.NOT_BOOKABLE, price }
}
