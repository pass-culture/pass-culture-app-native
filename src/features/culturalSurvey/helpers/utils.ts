import {
  CulturalSurveyQuestionEnum,
  CulturalSurveyQuestionsResponse,
  UserProfileResponse,
} from 'api/gen/api'
import { useSettingsContext } from 'features/auth/SettingsContext'

export const mapQuestionIdToPageTitle = (id: CulturalSurveyQuestionEnum | undefined) => {
  switch (id) {
    case CulturalSurveyQuestionEnum.SORTIES:
    case CulturalSurveyQuestionEnum.FESTIVALS:
    case CulturalSurveyQuestionEnum.SPECTACLES:
      return 'Tes sorties'
    case CulturalSurveyQuestionEnum.ACTIVITES:
      return 'Tes activitÃ©s'
    default:
      return ''
  }
}

export function useCulturalSurveyRoute() {
  const { data: settings } = useSettingsContext()

  // as long as ENABLE_CULTURAL_SURVEY is still used in the api to define if user should
  // fill cultural survey, we use enableNativeCulturalSurvey to know which cultural survey
  // (native or typeform) to show
  if (settings?.enableNativeCulturalSurvey) {
    return 'CulturalSurveyIntro'
  }
  return 'CulturalSurvey'
}

export function shouldShowCulturalSurvey(user?: UserProfileResponse) {
  if (!user) return false
  return user.needsToFillCulturalSurvey
}

export const createInitialQuestionsList = (
  culturalSurveyQuestions?: CulturalSurveyQuestionsResponse
) => {
  const subQuestions = [] as CulturalSurveyQuestionEnum[]
  const initialQuestionsList = [] as CulturalSurveyQuestionEnum[]

  if (!culturalSurveyQuestions?.questions) return initialQuestionsList

  culturalSurveyQuestions.questions.forEach((question) => {
    if (!subQuestions.includes(question.id)) {
      initialQuestionsList.push(question.id)
    }
    question.answers.forEach((answer) => {
      if (answer.sub_question && !subQuestions.includes(answer.sub_question)) {
        subQuestions.push(answer.sub_question)
      }
    })
  })
  return initialQuestionsList
}

export const addSubQuestionToQuestionsToDisplay = (
  subQuestionId: CulturalSurveyQuestionEnum,
  questionsToDisplay: CulturalSurveyQuestionEnum[]
) => {
  let updatedQuestionsToDisplay = [...questionsToDisplay]
  updatedQuestionsToDisplay = updatedQuestionsToDisplay.filter(
    (questionId) => questionId != subQuestionId
  )
  return updatedQuestionsToDisplay
}

export const removeSubQuestionsToDisplay = (
  subQuestionId: CulturalSurveyQuestionEnum,
  questionsToDisplay: CulturalSurveyQuestionEnum[]
) => {
  const updatedQuestionsToDisplay = [...questionsToDisplay]
  updatedQuestionsToDisplay.splice(1, 0, subQuestionId)
  return updatedQuestionsToDisplay
}
