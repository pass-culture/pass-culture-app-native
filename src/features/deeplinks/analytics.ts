import { RouteParams, AllNavParamList } from 'features/navigation/RootNavigator'
import { logConsultOfferFromDeeplink } from 'libs/analytics'

export const handleDeeplinkAnalytics = <Screen extends keyof AllNavParamList>(
  screen: Screen,
  params: RouteParams<AllNavParamList, Screen>
) => {
  if (screen === 'Offer') {
    logConsultOfferFromDeeplink((params as RouteParams<AllNavParamList, 'Offer'>).id)
  }
}
