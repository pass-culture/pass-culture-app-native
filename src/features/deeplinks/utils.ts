import { env } from 'libs/environment'

export const FIREBASE_DYNAMIC_LINK_URL = `https://${env.FIREBASE_DYNAMIC_LINK_DOMAIN}`

/**
 * @see https://firebase.google.com/docs/dynamic-links/create-manually
 */
export function getLongDynamicLinkURI() {
  return `apn=${env.ANDROID_APP_ID}&isi=${env.IOS_APP_STORE_ID}&ibi=${env.IOS_APP_ID}&efr=1`
}

/**
 * @see https://firebase.google.com/docs/dynamic-links/create-manually
 * @param deepLink The deeplink targeted screen
 * @param deepLinkParams Firebase additional params https://firebase.google.com/docs/dynamic-links/create-manually#parameters or ofl URL if a string. TODO: remove string support once webapp migration is complete
 */
export function generateLongFirebaseDynamicLink(
  deepLink: string,
  deepLinkParams?: Record<string, string>
) {
  let params = ''
  if (deepLinkParams) {
    Object.entries(deepLinkParams).forEach(([key, value]) => {
      params += `&${key}=${value}`
    })
  }
  return `${FIREBASE_DYNAMIC_LINK_URL}/?link=${encodeURIComponent(
    deepLink
  )}&${getLongDynamicLinkURI()}${params}`
}

export const isFirebaseDynamicLink = (url: string) => url.startsWith(FIREBASE_DYNAMIC_LINK_URL)
export const isFirebaseLongDynamicLink = (url: string) =>
  isFirebaseDynamicLink(url) && url.includes('?link=')
