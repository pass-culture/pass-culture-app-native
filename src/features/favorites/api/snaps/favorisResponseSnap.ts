import {
  ExpenseDomain,
  FavoriteResponse,
  PaginatedFavoritesResponse,
  SubcategoryIdEnum,
} from 'api/gen'

// humanizedId AHD3A
export const paginatedFavoritesResponseSnap: PaginatedFavoritesResponse = {
  page: 1,
  nbFavorites: 4,
  favorites: [
    {
      id: 380,
      offer: {
        id: 146193,
        name: 'Spectacle une seule date',
        expenseDomains: [ExpenseDomain.all],
        externalTicketOfficeUrl: null,
        image: null,
        coordinates: {
          latitude: 48.12108,
          longitude: -1.17896,
        },
        subcategoryId: SubcategoryIdEnum.SPECTACLE_ENREGISTRE,
        price: 0,
        startPrice: null,
        date: '2021-03-28T15:45:00',
        startDate: null,
        isReleased: true,
      },
    },
    {
      id: 381,
      offer: {
        id: 146112,
        name: 'Je ne sais pas ce que je dis',
        expenseDomains: [ExpenseDomain.all],
        externalTicketOfficeUrl: null,
        image: {
          credit: null,
          url: 'https://storage.gra.cloud.ovh.net/v1/AUTH_688df1e25bd84a48a3804e7fa8938085/storage-pc-dev/thumbs/mediations/CW8Q',
        },
        subcategoryId: SubcategoryIdEnum.CONFERENCE,
        coordinates: {
          latitude: 5.15839,
          longitude: -52.63741,
        },
        price: null,
        startPrice: 2400,
        date: null,
        startDate: '2021-03-02T20:00:00',
        isReleased: true,
      },
    },
    {
      id: 389,
      offer: {
        id: 146099,
        name: 'Un lit sous une rivière',
        expenseDomains: [ExpenseDomain.all],
        externalTicketOfficeUrl: null,
        image: {
          credit: null,
          url: 'https://storage.gra.cloud.ovh.net/v1/AUTH_688df1e25bd84a48a3804e7fa8938085/storage-pc-dev/thumbs/mediations/CWJQ',
        },
        subcategoryId: SubcategoryIdEnum.SPECTACLE_ENREGISTRE,
        coordinates: {
          latitude: 48.94538,
          longitude: 2.5029,
        },
        price: null,
        startPrice: 2700,
        date: null,
        startDate: '2021-03-04T20:00:00',
        isReleased: true,
      },
    },
    {
      id: 393,
      offer: {
        id: 146105,
        name: 'Un lit sous une rivière',
        expenseDomains: [ExpenseDomain.all],
        externalTicketOfficeUrl: null,
        image: {
          credit: null,
          url: 'https://storage.gra.cloud.ovh.net/v1/AUTH_688df1e25bd84a48a3804e7fa8938085/storage-pc-dev/thumbs/mediations/CWMA',
        },
        subcategoryId: SubcategoryIdEnum.SPECTACLE_ENREGISTRE,
        coordinates: {
          latitude: 48.9263,
          longitude: 2.49008,
        },
        price: null,
        startPrice: 2700,
        date: null,
        startDate: '2021-03-04T20:00:00',
        isReleased: true,
      },
    },
  ],
}

export const addFavoriteJsonResponseSnap: FavoriteResponse = {
  id: 1000,
  offer: {
    id: 10000,
    name: 'Spectacle de test',
    expenseDomains: [ExpenseDomain.all],
    externalTicketOfficeUrl: null,
    image: null,
    coordinates: {
      latitude: 48.12108,
      longitude: -1.17896,
    },
    subcategoryId: SubcategoryIdEnum.SPECTACLE_ENREGISTRE,
    price: 0,
    startPrice: null,
    date: '2021-04-01T12:00:00',
    startDate: null,
    isReleased: true,
  },
}
