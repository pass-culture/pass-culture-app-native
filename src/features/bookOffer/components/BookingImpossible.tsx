import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React, { useEffect } from 'react'
import styled from 'styled-components/native'

import { useBooking } from 'features/bookOffer/pages/BookingOfferWrapper'
import { useNotifyWebappLinkSent } from 'features/bookOffer/services/useNotifyWebappLinkSent'
import { useAddFavorite, useFavorite } from 'features/favorites/pages/useFavorites'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { ButtonTertiary } from 'ui/components/buttons/ButtonTertiary'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { SadFace } from 'ui/svg/icons/SadFace'
import { getSpacing, Spacer, Typo } from 'ui/theme'

export const BookingImpossible: React.FC = () => {
  const { bookingState, dismissModal } = useBooking()
  const { offerId } = bookingState
  const favorite = useFavorite({ offerId })
  const { navigate } = useNavigation<UseNavigationType>()
  const { showErrorSnackBar } = useSnackBarContext()
  const { mutate: notifyWebappLinkSent } = useNotifyWebappLinkSent()

  useEffect(() => {
    if (typeof offerId == 'undefined') return
    analytics.logBookingImpossibleiOS(offerId)
  }, [offerId])

  const { mutate: addFavorite } = useAddFavorite({
    onSuccess: () => {
      if (typeof offerId == 'undefined') return
      analytics.logHasAddedOfferToFavorites({ from: 'bookingimpossible', offerId })
      notifyWebappLinkSent(offerId)
    },
    onError: () => {
      showErrorSnackBar({
        message: t`L'offre n'a pas été ajoutée à tes favoris`,
        timeout: SNACK_BAR_TIME_OUT,
      })
    },
  })

  if (typeof offerId == 'undefined') return null

  const addToFavourite = () => {
    addFavorite({ offerId })
    dismissModal()
  }

  const navigateToOffer = () => {
    dismissModal()

    const from = 'bookingimpossible'
    analytics.logConsultOffer({ offerId, from })
    navigate('Offer', { id: offerId, from })
  }

  return (
    <Container>
      <GreySadFace />
      <Spacer.Column numberOfSpaces={6} />

      <Content>
        {t`Les conditions générales d'utilisation de l'App Store iOS ne permettent pas de réserver cette offre sur l'application.`}
      </Content>
      <Spacer.Column numberOfSpaces={6} />

      {favorite ? (
        <React.Fragment>
          <Content>{t`Rends-toi vite sur le site pass Culture afin de la réserver`}</Content>
          <Spacer.Column numberOfSpaces={6} />
          <ButtonPrimary wording={t`Voir le détail de l'offre`} onPress={navigateToOffer} />
        </React.Fragment>
      ) : (
        <React.Fragment>
          <Content>
            {t`Mets cette offre en favoris\u00a0: tu recevras une notification avec un lien pour la réserver sur notre application web\u00a0!`}
          </Content>
          <Spacer.Column numberOfSpaces={6} />
          <ButtonPrimary wording={t`Mettre en favoris`} onPress={addToFavourite} />
          <Spacer.Column numberOfSpaces={4} />
          <ButtonTertiary wording={t`Retourner à l'offre`} onPress={dismissModal} />
        </React.Fragment>
      )}
      <Spacer.Column numberOfSpaces={4} />
    </Container>
  )
}

const GreySadFace = styled(SadFace).attrs(({ theme }) => ({
  color: theme.colors.greyDark,
  size: theme.illustrations.sizes.medium,
}))``

const Container = styled.View({
  width: '100%',
  justifyContent: 'center',
  alignContent: 'center',
  alignItems: 'center',
})
const Content = styled(Typo.Body)({ textAlign: 'center', paddingHorizontal: getSpacing(6) })
