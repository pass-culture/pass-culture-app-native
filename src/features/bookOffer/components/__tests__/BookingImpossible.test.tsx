import { fireEvent, render } from '@testing-library/react-native'
import { rest } from 'msw'
import * as React from 'react'
import { QueryClient } from 'react-query'
import waitForExpect from 'wait-for-expect'

import { addFavoriteJsonResponseSnap } from 'features/favorites/api/snaps/favorisResponseSnap'
import { analytics } from 'libs/analytics'
import { env } from 'libs/environment'
import { EmptyResponse } from 'libs/fetch'
import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
import { server } from 'tests/server'
import { superFlushWithAct } from 'tests/utils'

import { BookingImpossible } from '../BookingImpossible'

const mockDismissModal = jest.fn()
jest.mock('features/bookOffer/pages/BookingOfferWrapper', () => ({
  useBooking: jest.fn(() => ({
    bookingState: {
      offerId: 20,
      stockId: undefined,
      step: undefined,
      quantity: undefined,
      date: undefined,
    },
    dismissModal: mockDismissModal,
  })),
}))

jest.mock('features/auth/AuthContext', () => ({
  useAuthContext: jest.fn(() => ({ isLoggedIn: true })),
}))

server.use(
  rest.post<EmptyResponse>(`${env.API_BASE_URL}/native/v1/me/favorites`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(addFavoriteJsonResponseSnap))
  )
)

describe('<BookingImpossible />', () => {
  it('should render', () => {
    const { toJSON } = render(reactQueryProviderHOC(<BookingImpossible />))
    expect(toJSON()).toMatchSnapshot()
  })

  it("should dismiss modal when clicking on 'Retourner à l'offre'", () => {
    const { getByText } = render(reactQueryProviderHOC(<BookingImpossible />))
    fireEvent.press(getByText("Retourner à l'offre"))
    expect(mockDismissModal).toHaveBeenCalledTimes(1)
  })

  it('should log analytics event when adding to favorites', async () => {
    const setup = (queryClient: QueryClient) => {
      queryClient.setQueryData('favorites', {
        favorites: [],
      })
    }

    const { getByText } = render(reactQueryProviderHOC(<BookingImpossible />, setup))

    fireEvent.press(getByText('Mettre en favoris'))
    await superFlushWithAct()

    await waitForExpect(() => {
      expect(analytics.logHasAddedOfferToFavorites).toHaveBeenCalledTimes(1)
      expect(analytics.logHasAddedOfferToFavorites).toHaveBeenCalledWith({
        from: 'offer',
        offerId: 20,
      })
    })
  })
})
