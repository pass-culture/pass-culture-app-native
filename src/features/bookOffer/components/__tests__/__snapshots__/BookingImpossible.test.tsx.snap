// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingImpossible /> should render 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    height={68}
    width={68}
  >
    <Text>
      undefined-SVG-Mock
    </Text>
  </View>
  <View
    numberOfSpaces={6}
    style={
      Array [
        Object {
          "height": 24,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 15,
          "lineHeight": 20,
          "paddingHorizontal": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Les conditions générales d'utilisation de l'App Store iOS ne permettent pas de réserver cette offre sur l'application.
  </Text>
  <View
    numberOfSpaces={6}
    style={
      Array [
        Object {
          "height": 24,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 15,
          "lineHeight": 20,
          "paddingHorizontal": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Ajoute cette offre à tes favoris et rends-toi vite sur le site pass Culture afin de la réserver.
  </Text>
  <View
    numberOfSpaces={6}
    style={
      Array [
        Object {
          "height": 24,
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eb0055",
        "borderRadius": 24,
        "borderWidth": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "maxWidth": 500,
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
        "width": "100%",
      }
    }
    testID="button-container"
  >
    <Text
      adjustsFontSizeToFit={false}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 15,
            "lineHeight": 20,
            "marginLeft": 5,
            "maxWidth": 650,
          },
        ]
      }
      testID="button-title"
      textColor="#ffffff"
    >
      Mettre en favoris
    </Text>
  </View>
  <View
    numberOfSpaces={4}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 24,
        "borderWidth": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "maxWidth": 500,
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
        "width": "100%",
      }
    }
    testID="button-container"
  >
    <Text
      adjustsFontSizeToFit={false}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#eb0055",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 15,
            "lineHeight": 20,
            "marginLeft": 5,
            "maxWidth": 650,
          },
        ]
      }
      testID="button-title"
      textColor="#eb0055"
    >
      Retourner à l'offre
    </Text>
  </View>
  <View
    numberOfSpaces={4}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />
</View>
`;
